extensions/rich_text_components/rte-output-display.component.spec.ts

```diff
@@ -175,9 +235,343 @@ describe('RTE display component', () => {
+    let node = document.createElement('p');
+    // eslint-disable-next-line oppia/no-inner-html
+    node.innerHTML = 'Hello world';
+    let expectedString = 'Hello world';
+    let readableText = component.getReadableTextFromNode(node.childNodes[0]);
+    expect(readableText).toBe(expectedString);
+  });
+
+  it('should be able to get readable text from non-interactive link node', () => {
+    let node = document.createElement('p');
+    // eslint-disable-next-line oppia/no-inner-html
+    node.innerHTML =
+      '<oppia-noninteractive-link ' +
+      'url-with-value="&quot;https://oppia.org&quot;" ' +
+      'text-with-value="&quot;Oppia&quot;">' +
+      '</oppia-noninteractive-link>';
+    let expectedString = 'Oppia';
+    let readableText = component.decodeHtmlEntities(
+      component.getReadableTextFromNode(node.childNodes[0])
+    );
+    expect(readableText).toBe(expectedString);
+  });
+
+  it('should be able to get readable text from non-interactive math node', () => {
+    let node = document.createElement('p');
+    // eslint-disable-next-line oppia/no-inner-html
+    node.innerHTML =
+      '<oppia-noninteractive-math math_content-with-value="' +
+      '{&amp;quot;raw_latex&amp;quot;:&amp;quot;x^2 + y^2 = z^2&amp;' +
+      'quot;,&amp;quot;svg_filename&amp;quot;:&amp;quot;' +
+      'mathImg_20250120_160257_55t4cfik6h_height_2d85_width_12d757_verti' +
+      'cal_0d715.svg&amp;quot;}" ng-version="11.2.14">' +
+      '</oppia-noninteractive-math>';
+    let expectedString = 'x^2 + y^2 = z^2';
+    let readableText = component.getReadableTextFromNode(node.childNodes[0]);
+    expect(readableText).toBe(expectedString);
+  });
+
+  it('should not change bg highlight color when prev and current element are same during voiceover playback', fakeAsync(() => {
+    let audioPlayingSpy = spyOn(audioplayerService, 'isPlaying');
+    audioPlayingSpy.and.returnValue(true);
+
+    component.highlighIdToSentenceText = {
+      highlightBlock1: 'Hello world',
+    };
+
+    let document = TestBed.inject(DOCUMENT);
+    let previousElement = document.createElement('p');
+    // eslint-disable-next-line oppia/no-inner-html
+    previousElement.innerHTML = 'Hello world';
+    previousElement.style.backgroundColor =
+      component.backgroundColorOfHighlightedSentence;
+
+    component.previousHighlightedElementId = 'highlightBlock1';
+
+    spyOn(
+      automaticVoiceoverHighlightService,
+      'getCurrentSentenceIdToHighlight'
+    ).and.returnValue('highlightBlock1');
+
+    spyOn(document, 'getElementById').and.returnValue(previousElement);
+
+    component.highlightSentenceDuringVoiceoverPlay();
+
+    let highlightedElement = document.getElementById('highlightBlock1');
+    expect((highlightedElement as HTMLElement).style.backgroundColor).toBe(
+      'rgb(243, 209, 64)'
+    );
+  }));
+
+  it('should highlight the current element are same during voiceover playback', fakeAsync(() => {
```

Comment:
Maybe change this to
```suggestion
  it('should highlight the current element and remove highlighting from previous element during voiceover playback', fakeAsync(() => {
```

Reply:
Done